### PR TITLE
add soaking for pull request

### PR DESCRIPTION
**Pull request by wyTrivail**
add a soaking test for pull request, which will be trigger by labelling

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -438,4 +438,38 @@ jobs:
         with:
           running_type: candidate
           opts: "-t=UploadCandidate -s=build/packages/.aoc-stack-test.yml"
-
+          
+  pr-soaking-test:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request' && github.event.label.name == 'soaking'
```
Comment (mxiamxia) [inline .github/workflows/CI.yml:444]: can we make label name to full "soaking test need" or something better? people can easy to understand from the PR?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -438,4 +438,38 @@ jobs:
...
+      
+      - name: Check out testing framework
+        uses: actions/checkout@v2
+        with:
+          repository: 'aws-observability/aws-otel-collector-test-framework'
+          ref: 'terraform'
+          
+      - name: Run testing suite
+        run: |
+          cd terraform/ec2 && terraform init && terraform apply -auto-approve -var-file="../testing-suites/soaking-ec2.tfvars" -var="aoc_version=${{ needs.e2etest-preparation.outputs.version }}"
```
Comment (mxiamxia) [inline .github/workflows/CI.yml:470]: Q: where to config the soaking test running duration?

File: .github/workflows/CI.yml (hunk 1 above)
Comment (wyTrivail) [reply .github/workflows/CI.yml:444]: okay, sounds good! let me change it

File: .github/workflows/CI.yml (hunk 2 above)
Comment (wyTrivail) [reply .github/workflows/CI.yml:470]: it's in the testing-suite of testing-framework: https://github.com/aws-observability/aws-otel-test-framework/blob/terraform/validator/src/main/resources/validations/alarm-pulling-validation.yml, please ignore the number now, i will enlarge them later

**Review by mxiamxia — APPROVED** [review]

**Pull request closed without merging**